From 2ede9f17adad4e9ebfbbf9d7da4f0df6101d5276 Mon Sep 17 00:00:00 2001
From: fatalc <cnfatal@gmail.com>
Date: Tue, 17 Feb 2026 02:46:35 +0800
Subject: [PATCH 5/8] Fix SDL2 and libpng builds for modern macOS/Xcode

- build-sdl.sh: Set IPHONEOS_DEPLOYMENT_TARGET=12.0 and
  ALWAYS_SEARCH_USER_PATHS=NO in xcodebuild to fix SDK 26.2
  compatibility (deployment target 5.1.1 too low, traditional
  headermap not supported)
- build-libpng.sh: Replace fp.h with math.h (fp.h removed in
  modern macOS SDKs)
---
 dependencies/scripts/build-libpng.sh | 3 +++
 dependencies/scripts/build-sdl.sh    | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/dependencies/scripts/build-libpng.sh b/dependencies/scripts/build-libpng.sh
index c85f88f..716f144 100755
--- a/dependencies/scripts/build-libpng.sh
+++ b/dependencies/scripts/build-libpng.sh
@@ -16,6 +16,9 @@ try mv libpng-$LIBPNG_VERSION $TMPROOT
 
 pushd $TMPROOT/libpng-$LIBPNG_VERSION
 
+# Fix for modern macOS/Xcode: fp.h is no longer available, use math.h
+sed -i.bak 's|include <fp\.h>|include <math.h>|' pngpriv.h
+
 echo 'Building libpng'
 try cp scripts/pnglibconf.h.prebuilt pnglibconf.h
 
diff --git a/dependencies/scripts/build-sdl.sh b/dependencies/scripts/build-sdl.sh
index 23acdc1..9010a77 100755
--- a/dependencies/scripts/build-sdl.sh
+++ b/dependencies/scripts/build-sdl.sh
@@ -19,8 +19,8 @@ try pushd $TMPROOT/SDL2-$SDL_VERSION
 echo 'Building SDL'
 
 pushd $TMPROOT/SDL2-$SDL_VERSION/Xcode-iOS/SDL
-try xcodebuild -project SDL.xcodeproj -target libSDL -configuration $RENIOSBUILDCONFIGURATION -sdk $SDKBASENAME$SDKVER -arch $RENIOSARCH clean
-try xcodebuild -project SDL.xcodeproj -target libSDL -configuration $RENIOSBUILDCONFIGURATION -sdk $SDKBASENAME$SDKVER -arch $RENIOSARCH
+try xcodebuild -project SDL.xcodeproj -target libSDL -configuration $RENIOSBUILDCONFIGURATION -sdk $SDKBASENAME$SDKVER -arch $RENIOSARCH IPHONEOS_DEPLOYMENT_TARGET=12.0 ALWAYS_SEARCH_USER_PATHS=NO clean
+try xcodebuild -project SDL.xcodeproj -target libSDL -configuration $RENIOSBUILDCONFIGURATION -sdk $SDKBASENAME$SDKVER -arch $RENIOSARCH IPHONEOS_DEPLOYMENT_TARGET=12.0 ALWAYS_SEARCH_USER_PATHS=NO
 popd
 
 popd
-- 
2.52.0

